src/tailwind.preset.js

```diff
@@ -1,32 +1,54 @@
+import plugin from 'tailwindcss/plugin.js';
+
 import colors from 'tailwindcss/colors.js';
 
 export default {
   theme: {
-    colors: {
-      transparent: 'transparent',
-      current: 'currentColor',
-      white: colors.white,
-      black: colors.black,
-      'grey-0': '#fafafa',
-      'grey-1': '#f2f2f2',
-      'grey-2': '#ececec',
-      'grey-3': '#dbdbdb',
-      'grey-4': '#a6a6a6',
-      'grey-5': '#9c9c9c',
-      'grey-6': '#737373',
-      'grey-7': '#595959',
-      'grey-8': '#3f3f3f',
-      'grey-9': '#202020',
-      brand: {
-        dark: '#84141e',
-        DEFAULT: '#bd1c2b',
+    extend: {
+      borderColor: {
+        DEFAULT: '#dbdbdb',
+      },
+      colors: {
+        transparent: 'transparent',
+        current: 'currentColor',
+        white: colors.white,
+        black: colors.black,
+        'grey-0': '#fafafa',
+        'grey-1': '#f2f2f2',
+        'grey-2': '#ececec',
+        'grey-3': '#dbdbdb',
+        'grey-4': '#a6a6a6',
+        'grey-5': '#9c9c9c',
+        'grey-6': '#737373',
+        'grey-7': '#595959',
+        'grey-8': '#3f3f3f',
+        'grey-9': '#202020',
+        green: {
+          success: '#00a36d',
+        },
+        yellow: {
+          notice: '#fbc168',
+        },
+        red: {
+          error: '#d93c3f',
+        },
+        brand: {
+          dark: '#84141e',
+          DEFAULT: '#bd1c2b',
+        },
+      },
+      spacing: {
+        'touch-minimum': '44px', // Equivalent to spacing 11; minimum touch-target size
       },
-      success: '#00a36d',
-      notice: '#fbc168',
-      error: '#d93c3f',
-    },
-    spacing: {
-      'touch-minimum': '44px', // Equivalent to spacing 11; minimum touch-target size
     },
   },
+  plugins: [
+    plugin(({ addVariant }) => {
+      // Add a custom variant such that the `theme-clean:` modifier is available
+      // for all tailwind utility classes. e.g. `.theme-clean:bg-white` would
+      // only apply (set the element's background color to white) if a parent
+      // element had the `.theme-clean` class.
+      addVariant('theme-clean', '.theme-clean &');
```

Comment:
The selector here refers to the `theme-clean` class that is set on the root element by the `HypothesisApp` component in the client? Does the clean theme need to be part of this project or could it be part of the client's configuration?

Reply:
> Does the clean theme need to be part of this project [...]

Are you asking in terms of mechanics (making sure Tailwind doesn't purge needed things) or strategy (it's weird to have some support for theming in this repo vs. some in the `client`)?

The first question is easy: we're fine mechanically to split it up. If we were to, say, use a class on a component in this project, let's call it `theme-clean:border-0` (turn off borders when the modifier `theme-clean` applies), Tailwind will see `theme-clean:border-0` and make sure it provides that utility class in the resulting stylesheet. That will look roughly like (this might be slightly off in a character or two):

```
.theme-clean .theme-clean\:border-0 {
  border-width: 0; // This is from Tailwind's `border-0` implementation
}
```

Tailwind doesn't care about where the outer `.theme-clean` comes from; adding the variant as is done here translates `theme-clean:border-0` into a selector based on the second argument passed to `addVariant`. 

As to the second question about strategy for the clean theme, there are many ways to shape that balloon into an animal. We could opt to do all of the `theme-clean` adaptations within within the client itself only, in fact we might want to. We could use the same `addVariant` there and extend the classes applied to shared components. So, instead of, say, a shared `Card` component here in this project adding clean-theme styling, we could use `Card` in the client like so:

```
<Card classes="theme-clean:border-0"> ... </Card>
```

In fact, that's likely a better approach. _But I don't want to break anything during the transition_, so propose we start here.

Reply:
Hi Lyza - I was thinking more about the strategy question. In other words, should the frontend-shared package know about the clean theme.

> In fact, that's likely a better approach. But I don't want to break anything during the transition, so propose we start here.

OK. I just wanted to make sure that we were considering where this logically belongs.